5.0 Draft/IDD/IDDs Service Registry/eu.arrowhead.service-discovery-http-json.yml

```diff
@@ -0,0 +1,631 @@
+    consumer registering it. This means that a consumer may be forbidden from registering or
+    deregistering a record for certain or all other service providers, even if it may register
+    and/or deregister records with itself as provider.
+
+    ## Provider Identifiers
+
+    Every service instance registered via this service interface _must_ be associated with a unique
+    identifier. In contexts where authentication is not used, providers _must_ chose their own
+    identifiers. Those identifiers _should_ remain consistent between restarts of those providers.
+    In contexts where authentication is used, the identifier _must_ be derived as follows: TODO.
+
+    ## Size Limits
+
+    As a mitigation against denial-of-service attacks, all implementations of this service
+    interface _should_ reject incoming requests that are larger than a predefined limit. That limit
+    _must not_ be smaller than 8192 bytes for each received request. An implementation _may_ also
+    limit the sizes of individual parts within each request. It _must_, however, receive request
+    URI:s up to 2816 bytes in size, as well as request payloads up to 4096 bytes in size.
+  version: 5.0.0-rc.1
+  # Additional information present in the IDD template
+  x-transfer-protocol: [HTTP, HTTPS]
+  x-encryption-method: to-add
+  x-encoding: JSON
+  x-compression: to-add
+  x-semantics: to-add
+  x-Ontology: to-add
+
+paths:
+
+  /echo:
+    get:  
+      summary: Endpoint to check the core system availability. 
+      description: Returns an OK to indicate that the core system is available. 
+      responses:
+        '200':
+          description: OK
+
+  /services:
+
+    get:
+      operationId: query
+      description: |
+        Searches for records matching any query parameters. If no query parameters are specified,
+        any records the consuming system are permitted to access may be provided in the response.
+
+        If the `limit` query parameter is used and given a value larger than 16, or if the `offset`
+        query parameter is used with any value, the operation is escalated from type `query` to type
+        `scan`. Permission to use the latter operation is intended only for consumers that need to
+        interact with larger number of service providers, that analyze larger numbers of records,
+        that list them for administrators or operators, or that have other special needs.
+
+        As long as the operation remains a plain `query`, it will always limit its response to 16
+        records, unless a smaller `limit` is specified. If escalated to type `scan`, default and
+        maximum limits depend on how the service is configured. The limits may vary depending on the
+        attributes of the consumer, such as by allowing for a higher limit if the consumer is an
+        administrator.
+
+        Any matching records a consumer are not allowed to access will be silently discarded before
```

Comment:
Again, this is authorization which is not in scope of the service registry in my opinion.

Reply:
How do you mean? Do you mean that service discovery should be available to all systems that can be authenticated with no restriction? Even if many implementations are going to be implemented this way, I don't see any reason for why we shouldn't allow for conditionally served service records.